rsocket-core/src/main/java/io/rsocket/fragmentation/FrameReassembler.java

```diff
@@ -25,143 +37,252 @@
  *     href="https://github.com/rsocket/rsocket/blob/master/Protocol.md#fragmentation-and-reassembly">Fragmentation
  *     and Reassembly</a>
  */
-final class FrameReassembler implements Disposable {
+final class FrameReassembler extends AtomicBoolean implements Disposable {
```

Comment:
why extend AtomicBoolean instead of a field?  memory?

Reply:
something I picked up from rxjava